File: Week_01/G20190343020229/26.删除排序数组中的重复项.py

class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        idx = 0
        jdx = 1
        while idx < len(nums) and jdx < len(nums):
            if nums[idx] != nums[jdx]:
                idx = idx + 1
                nums[idx] = nums[jdx]
            jdx = jdx + 1

        return len(nums[:idx+1])

File: Week_01/G20190343020149/26.删除排序数组中的重复项.js
/*
 * @lc app=leetcode.cn id=26 lang=javascript
 *
 * [26] 删除排序数组中的重复项
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    if( nums.length <= 1 ) return nums.length;

    let cursor = 0;
    nums.forEach( ( num , index)=>{
        if( nums[cursor] != num ){
            cursor++;
            if( cursor !== index ) nums[cursor] = num;
        }
    } )

    return cursor+1;
};
// @lc code=end

